code/game/atoms.dm

```diff
@@ -58,10 +58,23 @@
 
 	var/list/filter_data //For handling persistent filters
 
-	///Economy cost of item
+	/// Economy cost of item, 0 price items will not be sold and return when sent to CC trough cargo shuttle
 	var/custom_price
-	///Economy cost of item in premium vendor
+	/// Economy cost of item in premium vendor category (Export will use this if it exists even if custom price is defined)
 	var/custom_premium_price
+	/// Is this contraband? If so can only be exported if the supply console has access to contraband
+	var/is_contraband = FALSE
+	/// Maximum demand of the object type for exporting calculations
+	var/max_demand
+	/// Can this item be sold? TRUE by default. False means it will return when sent to CC via cargo shuttle (if delete_on_Sale_attempt is also false)
+	var/can_sell = TRUE
+	/// Should this get deleted even thought it didn't get sold?
+	var/delete_on_sale_attempt = FALSE
+	/// Tagger Price tag
+	var/price_tag
+	/// The weight class of an object. Used to determine tons of things, like if it's too cumbersome for you to drag, if it can fit in certain storage items, how long it takes to burn, and more. See _DEFINES/inventory.dm to see all weight classes.
+	var/w_class
```

Comment:
Really don't like this being pulled up to all atoms when its item specific. Implement a proc that generates the price calculation and use an algorithm that doesn't depend on w_class for non-items.